app/frontend/src/features/profile/HostingPreferenceForm.tsx

```diff
@@ -1,10 +1,234 @@
-import { makeStyles } from "@material-ui/core";
-import React from "react";
+import {
+  Checkbox,
+  FormControl,
+  FormControlLabel,
+  makeStyles,
+  Typography,
+} from "@material-ui/core";
+import { Autocomplete } from "@material-ui/lab";
+import { unwrapResult } from "@reduxjs/toolkit";
+import React, { useState } from "react";
+import { Controller, useForm, UseFormMethods } from "react-hook-form";
+import { updateHostingPreference } from "./index";
+import Alert from "../../components/Alert";
+import Button from "../../components/Button";
+import ProfileTextInput from "./ProfileTextInput";
+import { ProfileFormData } from "../../service/user";
+import { useAppDispatch, useTypedSelector } from "../../store";
+import { theme } from "../../theme";
+import { SmokingLocation } from "../../pb/api_pb";
+import { smokingLocationLabels } from "../../constants";
+
+interface HostingPreferenceCheckboxProps {
+  className: string;
+  defaultValue: boolean;
+  name: string;
+  label: string;
+  register: UseFormMethods<ProfileFormData>["register"];
+}
+
+function HostingPreferenceCheckbox({
```

Comment:
But can't a checkbox with a label be the same component as one without? I would have made an separate component.

Reply:
Probably with some kind of conditional rendering logic, as it would only need to render the `Checkbox` itself. Don't think it's worth the extra complexity for now though and I don't see the rush to move everything into components until obvious patterns/repetition emerge